cms/djangoapps/course_to_library_import/models.py

```diff
@@ -0,0 +1,74 @@
+"""
+Models for the course to library import app.
+"""
+
+from typing import Self
+
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from opaque_keys.edx.django.models import UsageKeyField
+
+from model_utils.models import TimeStampedModel
+
+from .data import CourseToLibraryImportStatus
+from .validators import validate_course_ids
+
+User = get_user_model()
+
+
+class CourseToLibraryImport(TimeStampedModel):
+    """
+    Represents a course import into a content library.
+    """
+
+    status = models.CharField(
+        max_length=100,
+        choices=CourseToLibraryImportStatus.choices,
+        default=CourseToLibraryImportStatus.PENDING
+    )
+    course_ids = models.TextField(
+        blank=False,
+        help_text=_('Whitespace-separated list of course keys for which to compute grades.'),
+        validators=[validate_course_ids]
+    )
+    library_key = models.CharField(max_length=100)
+    source_type = models.CharField(max_length=30)
+    metadata = models.JSONField(default=dict, blank=True, null=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return f'{self.course_ids} - {self.library_key}'
+
+    class Meta:
+        verbose_name = _('Course to Library Import')
+        verbose_name_plural = _('Course to Library Imports')
+
+    @classmethod
+    def get_by_id(cls, import_id: int) -> Self:
+        """
+        Get an import task by its ID.
+        """
+        return cls.objects.filter(id=import_id).first()
+
+
+class ComponentVersionImport(TimeStampedModel):
+    """
+    Represents a component version that has been imported into a content library.
+    This is a many-to-many relationship between a component version and a course to library import.
+    """
+
```

Comment:
Let us also add an FK to Component. That way, we can declare that `component` and `library_import` are unique_together.

I believe `source_usage_key` and `library_import` can also be declared as unique_together. 

In other words, we assert that for any given library import event, the source block is unique, and the destination block is unique.